### ISSUE TITLE
oneflow.gather_nd produces inconsistent values when slice indices are out of range

### ISSUE THREAD

**Issue by xxxyyyzzz12345** [Oneflow-Inc/oneflow#9013]
## Summary

oneflow.gather_nd somtimes produces inconsistent values when slice indices exceed the shape of input tensor. Should we consider an output of zero tensor or simply raise an index out of range exception?

## Code to reproduce bug
```python
for i in range(10):
    x = oneflow.rand(3,3)
    x.uniform_(-5,5)
    x_1 = x.clone()
    x_2 = x.clone()
    index = oneflow.tensor([[3],[4]],dtype=oneflow.int32)
    out_1 = oneflow.sum(oneflow.gather_nd(x_1,index))
    out_2 = oneflow.sum(oneflow.gather_nd(x_2,index))
    print(out_1-out_2)
```
##Output
```
tensor(0., dtype=oneflow.float32)
tensor(0., dtype=oneflow.float32)
tensor(0., dtype=oneflow.float32)
tensor(-4.2039e-45, dtype=oneflow.float32)
tensor(5.7079e+14, dtype=oneflow.float32)
tensor(4.2039e-45, dtype=oneflow.float32)
tensor(0., dtype=oneflow.float32)
tensor(-5.7417e+14, dtype=oneflow.float32)
tensor(5.7079e+14, dtype=oneflow.float32)
tensor(-1.1490e+15, dtype=oneflow.float32)
```
## System Information

- OneFlow version (run `python3 -m oneflow --doctor`):0.7.0+cu112

**Comment by ShiyuanResearch** [comment]
Thanks for your feedback. Please upgrade the stable version of OneFlow to 0.8.0 by `pip3 install oneflow -U`, or upgrade the OneFlow to nightly version by `python3 -m pip install --pre oneflow -f https://staging.oneflow.info/branch/master/cu112 -U`. This issue has been fixed in these releases, and the 0.7.0 version of OneFlow is no longer maintained.

**Comment by wyg1997** [comment]
We add indexing check in https://github.com/Oneflow-Inc/oneflow/pull/8938 . Please install the latest version and you will get the same error with your expectations.

```python
F20220829 10:05:00.654371 1967349 nd_index_slice_util.h:103] Check failed: nd_index[i] < dense_shape[i] && nd_index[i] >= -dense_shape[i] IndexError: index 3 is out of bounds for dimension 0 with size 3
*** Check failure stack trace: ***
    @     0x7f99ce7dfffc  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f99e8ae3005  oneflow::OffsetInSliceToOffsetInDense<>()
    @     0x7f99e8ae3378  oneflow::GatherNdKernel<>::Compute()
    @     0x7f99e93a05cc  oneflow::one::StatefulOpKernel::Compute()
    @     0x7f99e733d772  oneflow::vm::OpCallInstructionPolicy::Compute()
    @     0x7f99e733a64f  oneflow::vm::InstructionPolicy::ComputeIf()
    @     0x7f99e735c2e8  oneflow::vm::FuseInstructionPolicy::Compute()
    @     0x7f99e733a64f  oneflow::vm::InstructionPolicy::ComputeIf()
    @     0x7f99e733957f  oneflow::vm::EpStreamPolicyBase::Run()
    @     0x7f99e7342c41  oneflow::vm::ThreadCtx::TryReceiveAndRun()
    @     0x7f99e734e6c8  oneflow::(anonymous namespace)::WorkerLoop()
    @     0x7f99e7351ce1  std::thread::_State_impl<>::_M_run()
    @     0x7f99cdcd5de4  (unknown)
    @     0x7f9a61024609  start_thread
    @     0x7f9a6115e133  clone
```
